### PR TITLE
1.20

### PR DESCRIPTION
**Pull request by IMS212**
Seems to work perfectly, including testmod.

### CODE REVIEW

**Comment by emilyploszaj** [comment]
Hey- don't want to leave this unresponded to for too long. I appreciate this update and assuming it looks good plan to merge it, but currently don't have the time to review it, I'll try to get to it in the next couple of days, thanks for your contribution :slightly_smiling_face:

**Comment by CammiePone** [comment]
Fingers crossed it's all good! I just tried porting Icarus to 1.20 but some mixins in Trinkets failed because Mojang made some changes.

**Review by Noaaan — APPROVED** [review]

**Review by gliscowo — APPROVED** [review]

**Review by Dragon-Seeker — APPROVED** [review]

**Review by BasiqueEvangelist — APPROVED** [review]

**Review by chyzman — APPROVED** [review]

**Review by AlphaMode — APPROVED** [review]

**Review by DaFuqs — APPROVED** [review]

**Review by Azzyypaaras — APPROVED** [review]

**Review by CodexAdrian — APPROVED** [review]

**Review by LordDeatHunter — APPROVED** [review]

File: src/main/java/dev/emi/trinkets/mixin/CreativeInventoryScreenMixin.java [hunk 1]
```diff
@@ -51,7 +52,7 @@ private int size(DefaultedList<ItemStack> list) {
 
 	@Inject(at = @At("HEAD"), method = "setSelectedTab")
 	private void setSelectedTab(ItemGroup g, CallbackInfo info) {
-		if (g != ItemGroups.INVENTORY) {
+		if (g.getType() == ItemGroup.Type.INVENTORY) {
```
Comment (TropheusJ) [inline src/main/java/dev/emi/trinkets/mixin/CreativeInventoryScreenMixin.java:55]: inverted?

File: build.gradle [hunk 2]
```diff
@@ -31,6 +28,18 @@ sourceSets {
 	}
 }
 
+loom {
+	accessWidenerPath = file("src/main/resources/trinkets.accesswidener")
+
+	runs {
+		testmodClient {
+			client()
+			ideConfigGenerated project.rootProject == project
```
Comment (modmuss50) [inline build.gradle:37]:
```suggestion
```
Little nit pick: No need to set `ideConfigGenerated` to `project.rootProject == project` as this is the default behaviour. In the root project its worth setting `ideConfigGenerated` if you want to disable it.
😜

**Review by modmuss50 — APPROVED** [review]

**Review by Benonardo — APPROVED** [review]

**Review by enjarai — APPROVED** [review]

File: src/main/java/dev/emi/trinkets/mixin/CreativeInventoryScreenMixin.java (hunk 1 above)
Comment (IMS212) [reply src/main/java/dev/emi/trinkets/mixin/CreativeInventoryScreenMixin.java:55]: Not sure how I didn't notice this, will fix.

**Review by AlphaMode — APPROVED** [review]

**Review by chyzman — APPROVED** [review]
i liek

**Review by kikugie — APPROVED** [review]

**Review by X-13Shadow — APPROVED** [review]

**Review by enjarai — APPROVED** [review]

**Review by TropheusJ — APPROVED** [review]

**Review by Dragon-Seeker — APPROVED** [review]

**Review by JanCantCode — APPROVED** [review]
great!

**Review by Deftu — APPROVED** [review]

File: build.gradle [hunk 3]
```diff
@@ -31,6 +28,17 @@ sourceSets {
 	}
 }
 
+loom {
+	accessWidenerPath = file("src/main/resources/trinkets.accesswidener")
+
```
Comment (mineblock11) [inline build.gradle:33]: I dont like the look of this whitespace, it's too empty.

**Review by mineblock11 — APPROVED** [review]

**Comment by AlphaMode** [comment]
This is truly a new age in modding

File: gradle.properties [hunk 4]
```diff
@@ -1,14 +1,14 @@
 # Done to increase the memory available to gradle.
```
Comment (chyzman) [inline gradle.properties:1]: already removed the rest of the comments here so why is this one still here

**Review by chyzman — CHANGES_REQUESTED** [review]

**Review by Trent2087 — APPROVED** [review]

**Review by SylvKT — APPROVED** [review]
needs more reflection and java sandbox escape hacks but otherwise good
can't wait for the so-called "new age of modding" these liberals are talking about

**Review by Benjamin-Norton — APPROVED** [review]

**Review by fzzyhmstrs — APPROVED** [review]

**Review by uku3lig — APPROVED** [review]

**Review by birbe — APPROVED** [review]
lgtm

**Review by Minenash — APPROVED** [review]
Help, I'm being forced to approve, I don't approve!!

**Review by Batku — APPROVED** [review]
lgtm -batku

**Review by Batku — APPROVED** [review]
lgtm -batku

**Review by birbe — APPROVED** [review]
still lgtm

**Review by Technici4n — APPROVED** [review]
We live in a society

**Review by TheCurle — APPROVED** [review]
:shipit:

**Review by baileyholl — APPROVED** [review]

**Review by Dragon-Seeker — APPROVED** [review]

**Comment by Noaaan** [comment]
While someone might see these PR reviews and instantly label it as mindless trolling... This is not what I had in mind when reviewing this.

Today we were directly attacked in the form of malware, yet together with the fast acting of many community members, including Emily, we were able to do an incredible amount of work to prevent many users from being harmed and affected by it.

We are coming together as a community, to show an act of solidarity against these perpetrators. Them trying to harm us at the worst possible time, just before a major update day.

Emily mentioned that she would not be able to port Trinkets, an important mod for many of us, due to her work in keeping everyone safe. Because of this, I have taken it upon myself, with some help from glisco, Blod, chyz, alphamode and a ***few*** other reviewers to help out in getting it ported in time. 

Consider this a love letter to the entire community. Thank you all for your wonderful work. Lets make 1.20 and the next years going forward be the best they can be!

(We are actually making some bug fixes by the way, this is not completely baseless)

**Review by Noaaan — APPROVED** [review]

**Comment by emilyploszaj** [comment]
I intended to have this published but I am busy organizing and attempting to mitigate the `fractureiser` incident. I can't stop you from meming but just letting those know who care that I will be taking a look at this when my time isn't spent elsewhere.

**Review by quat1024 — APPROVED** [review]

**Review by EdwardJXLi — APPROVED** [review]

**Review by valoeghese — APPROVED** [review]

**Review by ramidzkh — APPROVED** [review]

**Review by triphora — APPROVED** [review]

**Review by maximumpower55 — APPROVED** [review]
lgtm

**Review by TheZstriker1 — APPROVED** [review]

**Review by isXander — APPROVED** [review]

**Review by chronosacaria — APPROVED** [review]
Emily, please know that myself and the whole of the Timefall Development Team are beyond appreciative for everything that you're doing with mitigating `fractureiser`. You stepped up and helped organise a team effort that reached across so many lines. You are truly admirable!

(btw, I actually looked at the code; looks good)

**Review by PotatoPresident — APPROVED** [review]

**Review by leasomeone — APPROVED** [review]

**Comment by leasomeone** [comment]
dayum the whole team ported to 1.20 <3

**Review by MysticFen — APPROVED** [review]

**Review by emilyploszaj — CHANGES_REQUESTED** [review]
Content looks good, for style though every instance of `info` was replaced with `ci` in modified method bodies, that needs to be reverted to merge.

**Review by emilyploszaj — APPROVED** [review]

**Review by jaskarth — COMMENTED** [review]
LGTM

**Comment by AlphaMode** [comment]
Rejoice! This legendary PR has finally been merged

**Pull request merged**